example/index.js

```diff
@@ -38,6 +38,18 @@ q.splice(2, 0, function (cb) {
   cb()
 })
 
+// jobs can send a result to the 'success' event
+function asyncJob () {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve('we waited 75ms'), 75)
+  })
+}
+q.push(function (cb) {
+  asyncJob()
+    .then((result) => cb(null, result))
+    .catch((error) => cb(error))
+})
```

Comment:
I think you may be able to do `q.push(asyncJob)` here

Reply:
Yes I suppose you could do that. If we want to demonstrate how to pass `results` through the `callback()` method, then maybe we need some other example. Consider the following

```suggestion
}) q.push( function ( cb ) { 
    const result = 'this is a long stream of data'
    cb( null, result )
    })
```

Reply:
Yes I agree, you could just return a promise. If we want to demonstrate how to pass a `result` into a `callback()` then maybe a different approach would work for you.

```suggestion
  const result = 'this is a long stream of data'
  cb(null, result)
})
```